packages/siri-sx-generator/index.ts

```diff
@@ -1,16 +1,45 @@
-import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { GetObjectCommand, S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
 import { S3Event } from "aws-lambda";
+import { toXML } from "jstoxml";
 
 const s3Client = new S3Client({ region: "eu-west-2" });
 
+const uploadToS3 = async (data: string, filename: string) => {
+    // Unique bucket name
+    const bucketName = process.env.SIRI_SX_BUCKET_NAME || "";
+    // Name for uploaded object
+    const keyName = filename;
+
+    const putCommand: PutObjectCommand = new PutObjectCommand({
+        Bucket: bucketName,
+        Key: keyName,
+        Body: data,
+    });
+
+    await s3Client.send(putCommand);
+    console.log("Successfully uploaded data to " + bucketName + "/" + keyName);
+};
+
 export const main = async (event: S3Event): Promise<void> => {
     const bucketName = event.Records[0].s3.bucket.name || "";
     const objectKey = event.Records[0].s3.object.key || "";
 
     const params = { Bucket: bucketName, Key: objectKey };
     const response = await s3Client.send(new GetObjectCommand(params));
-    const data = response.Body?.toString() || "";
+    const data = (await response.Body?.transformToString()) || "";
```

Comment:
How come we're using transformToString() instead of toString()?

Reply:
@RobCattonKPMG That's the function exposed to convert the stream returned by the AWS SDK into a string, `toString` won't work here

Reply:
Fair enough! Haven't seen it before